lib/index.js

```diff
@@ -2448,17 +2448,26 @@ exposeMethod('last');
  * _(filenames).through(jsonParser).map(function (obj) {
  *     // ...
  * });
+ *
+ * // All errors will be propagated as Highland errors
+ * _(['zz{"a": 1}']).through(jsonParser);
+ * // => [SyntaxError: Unexpected token z]
  */
 
 Stream.prototype.through = function (target) {
+    var s, writeErr = _.curry(function (xs, err) {
```

Comment:
No need to do this convoluted curry thing. You already have a closure with the variable `s`. Just define a

```javascript
function writeErr(err) {
    s.write(new StreamError(err));
}
```